### PR TITLE
Add real methods to person class

### PR DESCRIPTION
**Pull request by ondenman**
Adds real methods to class to replace methods dynamically generated in Entity.

Addresses issue: https://github.com/everypolitician/everypolitician-popolo/issues/54

### CODE REVIEW

**Comment by ondenman** [comment]
Not sure whether .contact() is supposed to be public facing. (https://github.com/everypolitician/everypolitician-popolo/blob/54-add-real-methods-to-person-class/test/everypolitician/popolo/person_test.rb) It’s called in the test: https://github.com/everypolitician/everypolitician-popolo/blob/54-add-real-methods-to-person-class/test/everypolitician/popolo/person_test.rb but I assume it’s meant to be private.

**Review by tmtmtmtm — CHANGES_REQUESTED** [review]
- This still has "WIP" in the title
- I'm a little uncomfortable with this removing all the existing tests. Adding _new_ tests against live data is a very good thing, but we're losing tests now (for example there are now only `assert_nil` tests for honorifics — i.e. none that actually show that the method works in cases where someone _does_ have those.
- You've made methods private that are used externally, e.g. in https://github.com/everypolitician/everypolitician-data/blob/master/rake_build/generate_final_csvs.rb

File: lib/everypolitician/popolo/person.rb [hunk 1]
```diff
@@ -3,22 +3,64 @@ module Popolo
...
 
-      def links
-        document.fetch(:links, [])
+      def given_name
+        document[:given_name]
       end
 
-      def link(type)
-        links.find(-> { {} }) { |i| i[:note] == type }[:url]
+      def identifiers
```
Comment (tmtmtmtm) [inline lib/everypolitician/popolo/person.rb:14]: This one already lives in the parent class.

File: lib/everypolitician/popolo/person.rb [hunk 2]
```diff
@@ -3,22 +3,64 @@ module Popolo
...
+        document[:identifiers]
       end
 
-      def contact_details
-        document.fetch(:contact_details, [])
+      def images
+        document[:images]
```
Comment (tmtmtmtm) [inline lib/everypolitician/popolo/person.rb:19]: Things that return lists are going to need to a little more complex than this, I suspect.

File: lib/everypolitician/popolo/person.rb [hunk 3]
```diff
@@ -42,7 +84,19 @@ def wikidata
       end
 
       def sort_name
-        name
+        document[:name]
```
Comment (tmtmtmtm) [inline lib/everypolitician/popolo/person.rb:87]: This one will need to also cope with the case of someone actually having a `sort_name` field set.

File: lib/everypolitician/popolo/person.rb [hunk 4]
```diff
@@ -42,7 +84,19 @@ def wikidata
...
+      end
+
+      def memberships
+        popolo.memberships.where(person_id: id)
+      end
+
+      def identifier(scheme_name)
+        identifiers.find { |i| i[:scheme] == scheme_name }[:identifier] rescue nil
```
Comment (tmtmtmtm) [inline lib/everypolitician/popolo/person.rb:95]: This one already exists in the parent class.

File: lib/everypolitician/popolo/person.rb [hunk 5]
```diff
@@ -57,8 +111,10 @@ def name_at(date)
         at_date.first[:name]
       end
 
-      def memberships
-        popolo.memberships.where(person_id: id)
+      private
+
+      def link(type)
```
Comment (tmtmtmtm) [inline lib/everypolitician/popolo/person.rb:116]: Why is this one private?

**Review by tmtmtmtm — CHANGES_REQUESTED** [review]
I'm still also very scared by the changes to the existing test. Ideally that test file should have zero changes so we can see that you're introducing any breaking behaviour — all the tests you're adding against real data should exist in a new test file. 

I think you also need negative tests for as many cases as possible (i.e where someone doesn't _have_ a `gender` field set, etc.)

**Comment by tmtmtmtm** [comment]
@ondenman feel free to break this (and the parallel one for Organization) up into multiple PRs. There are a lot of issues here, so feel free to pick off the new methods in bunches, rather than trying to add so many in one shot (especially as you're not being complete anyway)

**Comment by ondenman** [comment]
Started a new branch and PR: https://github.com/everypolitician/everypolitician-popolo/pull/82 and https://github.com/everypolitician/everypolitician-popolo/pull/83

**Pull request closed without merging**